platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -448,6 +448,37 @@ MGL_EXPORT
 
 - (void)putResourceWithUrl:(NSURL *)url data:(NSData *)data modified:(nullable NSDate *)modified expires:(nullable NSDate *)expires etag:(nullable NSString *)etag mustRevalidate:(BOOL)mustRevalidate __attribute__((deprecated("", "-preloadData:forURL:modificationDate:expirationDate:eTag:mustRevalidate:")));
 
+
+/**
+A block to be called once the data  has been preloaded.
+*/
+typedef void (^MGLOfflinePreloadDataCompletionHandler)(void);
```

Comment:
Where possible, a completion handler should accept some contextual information as parameters so that the developer can reuse their completion block code for multiple calls to the method. In this case, it would probably be helpful to pass in the URL and maybe the data.

Reply:
done